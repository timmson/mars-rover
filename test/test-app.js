import React from "react"
import renderer, {act} from "react-test-renderer"

import App from "../src/app"


const mockWindow = {
	addEventListener: () => {
	}
}

const mockDocument = {
	documentElement: {
		clientWidth: 0,
		clientHeight: 0
	}
}


describe("App should", () => {

	test("render", () => {
		let component
		act(() => {
			component = renderer.create(<App window={mockWindow} document={mockDocument}/>)
		})
		expect(component.toJSON()).toMatchSnapshot()
		component.unmount()
	})

})