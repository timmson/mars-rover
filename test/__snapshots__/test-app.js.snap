// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light container mb-5"
  >
    <div
      className="row w-50"
    >
      <div
        className="col-3 text-center"
      >
        Control:Â 
        <img
          alt="wasd"
          src="img/wasd.png"
        />
      </div>
      <div
        className="col-9 text-center"
      >
        <p>
          Tracking: 
          1
          , [
          0
          ,
          0
          ]
        </p>
      </div>
    </div>
  </nav>,
  <div
    className="rover"
    style={
      Object {
        "left": "-64px",
        "top": "306px",
        "transform": "rotate(90deg) scaleX(-1)",
      }
    }
  >
    <img
      alt="rover"
      src="img/rover.png"
    />
  </div>,
  <div
    className="container mars"
    style={
      Object {
        "background": "url(img/bg.png)",
      }
    }
  >
    <div
      className="row"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <div
        className="col-6 border border-1 border-dark text-start align-text-bottom"
      />
      <div
        className="col-6 border border-1 border-dark text-start align-text-top"
      />
    </div>
    <div
      className="row"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <div
        className="col-6 border border-1 border-dark text-end align-text-bottom"
      />
      <div
        className="col-6 border border-1 border-dark text-end align-text-top"
      />
    </div>
  </div>,
]
`;
